### PR TITLE
E2E tests: Define ansible roles

### PR DESCRIPTION
**Pull request by BelenValdivia**
|Related issue|
|-------------|
|     https://github.com/wazuh/wazuh-qa/issues/3144        |

## Description
Added some roles to encapsulate common tasks

### Added

The Ansible roles added are: 

- manage_wazuh: 

1. restart_wazuh.yaml

- manage_alerts:

1. get alert json
2. truncate alert json

- manage_wazuh_configurations

1. configure ossec.conf
2. configure local rules

### Updated

-Updated all tests to run with roles

---

### Audit
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386060/R1-test-audit-roles.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386061/R2-test-audit-roles.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386063/R3-test-audit-roles.zip)  |         |         | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### AWS
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢 ](https://github.com/wazuh/wazuh-qa/files/9386112/r1-aws.zip)[🟢](https://github.com/wazuh/wazuh-qa/files/9386113/r2-aws.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386115/r3-aws.zip)  |         |    9595bc9c5d1484f172d4d199ae69050fba23ca9e     | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Brute Force
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386116/r1-brute-force.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386117/r2-brute-force.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386118/r3-brute-force.zip)  |         |    9595bc9c5d1484f172d4d199ae69050fba23ca9e     | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Detecting suspicious binaries
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386120/r1-suspicious-binaries.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386119/r2-suspicious-binaries.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386121/r3-suspicious-binaries.zip)  |         |  9595bc9c5d1484f172d4d199ae69050fba23ca9e       | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Docker monitoring
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386123/r1-docker.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386124/r2-docker.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386126/r3-docker.zip)  |         |   9595bc9c5d1484f172d4d199ae69050fba23ca9e      | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Emotet
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386128/r1-emotet.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386129/r2-emotet.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386130/r3-emotet.zip)  |         |  9595bc9c5d1484f172d4d199ae69050fba23ca9e       | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Fim
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386131/r1-fim.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386132/r2-fim.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386133/r3-fim.zip)  |         |  9595bc9c5d1484f172d4d199ae69050fba23ca9e       | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### IP Reputation
| Tester         | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386134/r1-ip-reputation.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386135/r2-ip-reputation.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386136/r3-ip-reputation.zip)  |         |  9595bc9c5d1484f172d4d199ae69050fba23ca9e       | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Netcat
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386137/r1-netcat.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386138/r2-netcat.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386139/r3-netcat.zip)  |         |   9595bc9c5d1484f172d4d199ae69050fba23ca9e      | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### OSQuery
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386140/r1-osquery.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386141/r2-osquery.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386142/r3-osquery.zip)  |         | 9595bc9c5d1484f172d4d199ae69050fba23ca9e        | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Shellshock attack
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386143/r1-shellshock.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386144/r2-shellshock.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386145/r3-shellshock.zip)  |         |  9595bc9c5d1484f172d4d199ae69050fba23ca9e       | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Slack integration
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢 ](https://github.com/wazuh/wazuh-qa/files/9386149/r1-slack.zip)[🟢](https://github.com/wazuh/wazuh-qa/files/9386147/r2-slack.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386148/r3-slack.zip)  |         |  9595bc9c5d1484f172d4d199ae69050fba23ca9e       | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### SQL Injection
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386151/r1-sql-injection.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386152/r2-sql-injection.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386153/r3-sql-injection.zip)  |         |  9595bc9c5d1484f172d4d199ae69050fba23ca9e       | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Suricata 
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9396631/R1-test-suricata.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9396632/R2-test-suricata.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9396633/R3-test-suricata.zip)  |         |   9595bc9c5d1484f172d4d199ae69050fba23ca9e      | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Virustotal integration
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386154/r1-virustotal.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386155/r2-virustotal.zip)[ 🟢](https://github.com/wazuh/wazuh-qa/files/9386156/r3-virustotal.zip)  |         |    9595bc9c5d1484f172d4d199ae69050fba23ca9e     | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Vulnerability detector
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386158/r1-vulnerability-detector.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386159/r2-vulnerability-detector.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386160/r3-vulnerability-detector.zip)  |         |   9595bc9c5d1484f172d4d199ae69050fba23ca9e      | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Windows Defender
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386161/r1-windows-defender.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386162/r2-windows-defender.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386163/r3-windows-defender.zip)  |         |  9595bc9c5d1484f172d4d199ae69050fba23ca9e       | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### Yara integration
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia  (Developer)  |           | ⚫⚫⚫ | [🟢](https://github.com/wazuh/wazuh-qa/files/9386164/r1-yara.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386165/r2-yara.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9386166/r3-yara.zip)  |         |    9595bc9c5d1484f172d4d199ae69050fba23ca9e     | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### CODE REVIEW

**Review by juliamagan — CHANGES_REQUESTED** [review]
GJ

File: tests/end_to_end/roles/manage_wazuh/tasks/restart_wazuh.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+- name: Restart agent service on linux
+  become: true
+  service:
+    name: wazuh-agent
+    state: restarted
+  when: (os == 'linux' and 'agent' in wazuh_info.stdout)
+
+- name: Restart wazuh on Windows
+  win_shell: |
+    "Get-Service -Name WazuhSvc -ErrorAction SilentlyContinue | Restart-Service -ErrorAction SilentlyContinue"
```
[the remark this answers is not in the document]
Comment (BelenValdivia) [reply tests/end_to_end/roles/manage_wazuh/tasks/restart_wazuh.yaml]: Changed for `net stop Wazuh` and `net start Wazuh`. Fails when there is a wrong configuration

**Review by juliamagan — APPROVED** [review]
LGTM

File: tests/end_to_end/test_basic_cases/test_vulnerability_detector/test_vulnerability_detector_windows/data/playbooks/configuration.yaml [hunk 2]
```diff
@@ -38,12 +45,12 @@
       shell: echo "" > /var/ossec/logs/ossec.log
       become: true
 
-    - name: Enable vulnerability detector module
-      become: true
-      blockinfile:
-        path: /var/ossec/etc/ossec.conf
-        insertbefore: </ossec_config>
-        block: |
+    - name: Enabled vulnerability detector module
```
Comment (jmv74211) [inline tests/end_to_end/test_basic_cases/test_vulnerability_detector/test_vulnerability_detector_windows/data/playbooks/configuration.yaml:48]:
```suggestion
    - name: Enable vulnerability detector module
```

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/end_to_end/test_basic_cases/test_vulnerability_detector/test_vulnerability_detector_windows/data/playbooks/configuration.yaml (hunk 2 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_basic_cases/test_vulnerability_detector/test_vulnerability_detector_windows/data/playbooks/configuration.yaml]: Done in 4bf678883f2085dd60f5bd3ab01f0f5089dd1e33

File: tests/end_to_end/roles/manage_wazuh_configurations/tasks/write_wazuh_config.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# REQUIRED VARIABLES
+# -------------------
+#
+# GENERIC:
+#  - (String) os: Target operating system
```
[the remark this answers is not in the document]
Comment (BelenValdivia) [reply tests/end_to_end/roles/manage_wazuh_configurations/tasks/write_wazuh_config.yaml]: Done in 42f7020c28a0f00494a235c1e2b01b20c196873d

File: tests/end_to_end/roles/manage_wazuh_configurations/tasks/write_local_decoders.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+- name: Configure local decoders
```
[the remark this answers is not in the document]
Comment (BelenValdivia) [reply tests/end_to_end/roles/manage_wazuh_configurations/tasks/write_local_decoders.yaml]: Done in 42f7020c28a0f00494a235c1e2b01b20c196873d

File: tests/end_to_end/roles/manage_wazuh_configurations/tasks/write_local_rules.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+- name: Configure local rules
```
[the remark this answers is not in the document]
Comment (BelenValdivia) [reply tests/end_to_end/roles/manage_wazuh_configurations/tasks/write_local_rules.yaml]: Done in 42f7020c28a0f00494a235c1e2b01b20c196873d

File: tests/end_to_end/conftest.py [hunk 6]
```diff
@@ -152,3 +161,12 @@ def pytest_addoption(parser):
         type=str,
         help='Inventory path',
     )
+
+    parser.addoption(
+        '--roles_path',
+        action='store',
+        metavar='ROLES_PATH',
+        default=os.path.join(suite_path, 'roles'),
```
[the remark this answers is not in the document]
Comment (BelenValdivia) [reply tests/end_to_end/conftest.py]: Done in f580bbecefa466e7e95c34c056bbf6686f9cfbcf

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**